statocles-site/blog/2024/12/25/jwst/index.markdown

```diff
@@ -0,0 +1,139 @@
+
+    imag $jwst, -0.05, 0.1;
+    @dims = dims($jwst);
+    print join(" ", @dims),"\n";
+
+You can see from the output this is a 2D image of dimensions 435x31 and here is what it looks like:
+
+![2D image](https://github.com/user-attachments/assets/7db5ef7f-e368-4413-89d2-d4aaa76c3246)
+
+The horizontal bright trace down the middle is the spectrum of the galaxy. (The sharp cutoff on the left is due to the extreme redshift of the galaxy as we will see below, so is a genuine astrophysical effect and not an artefact). The dark bands either side are negative traces which _are_ an artefact of the  data reduction pipeline, specifically`sky subtraction via nodding the telescope'. These should be ignored.
+
+## Dimension tricks
+
+Standard procedure would be to sum along the columns of the image to produce a 435 pixel spectrum, summing up the flux of the galaxy along the slit. We could use the `sumover()` function, but `$jwst->sumover` would _sum_ along the rows and produce a 31 pixel ndarray which is not what we want. 'Reduction' operations like `sumover(), average(), medover() `etc. apply along the first axis and make a result with one dimension fewer. To get what we want we have to transpose the image, i.e. swap the two axes. There are various ways of doing this in PDL, the simplest is to use the `mv() `function as in `$jwst->mv(1,0)->sumover()`. This will move dimension 1 to the place of dimension 0, i.e. transposing the 2D ndarray. You can see using `dims() `that the result is a 435 element 1D ndarray.
+
+However in this case the resulting spectrum is hot garbage because by summing along the _entire_ column one adds in the black trace artefacts.
+
+## Broadcasting
+
+We need to only extract the central few rows. Code like
+`$jwst->slice(':,13:17')->mv(1,0)->sumover`  would work, but we can find a more optimal solution and demonstrate 'broadcasting' at the same time! First look at this code:
+
+    # Make a gaussian extraction profile
+
+    $x = xvals($dims[1]);
+    $fwhm = 2.1; # pix
+    $gaussian = 0.04*exp(-0.5*(($x-15)/($fwhm/2.35))**2);
+    line $x, $gaussian; # Plot
+
+This code generates a 31 element gaussian function along the slit axis. Here is what `$gaussian` looks like:
+
+![image](https://github.com/user-attachments/assets/7c3bf9aa-b028-4511-83f2-e460fd0afcd8)
+
+(It does not look smooth as it is evaluated on a coarse pixel grid and is only a few pixels wide). If we can extract along the slit weighting by this gaussian we can get two things for the price of one: (1) eliminate the signal from the dark artefacts and (2) _optimally_ weight the spectrum according to its light distribution along the slit (this arises from the shape of the galaxy) to get the most signal:noise.
+
+Now we can do some _broadcasting_ and some _reduction_:
+
+    $profile_weighted = $jwst * $gaussian->clump(0); # Multiply through
+    $spec1d = $profile_weighted->mv(1,0)->sumover; # Extract by summing along columns
```

Comment:
`t`

Reply:
are we also changing `$jwst->slice(':,13:17')->mv(1,0)->sumover` to `t` on line 76?

Reply:
Just the `mv` bit, but yes.